src/main/java/org/terasology/climateConditions/igloo/IglooProvider.java

```diff
@@ -0,0 +1,102 @@
+import org.terasology.core.world.generator.facets.BiomeFacet;
+import org.terasology.math.TeraMath;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.utilities.procedural.Noise;
+import org.terasology.utilities.procedural.WhiteNoise;
+import org.terasology.world.generation.Border3D;
+import org.terasology.world.generation.Facet;
+import org.terasology.world.generation.FacetBorder;
+import org.terasology.world.generation.FacetProviderPlugin;
+import org.terasology.world.generation.GeneratingRegion;
+import org.terasology.world.generation.Produces;
+import org.terasology.world.generation.Requires;
+import org.terasology.world.generation.facets.SeaLevelFacet;
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+import org.terasology.world.generation.facets.base.BaseFieldFacet2D;
+import org.terasology.world.generator.plugin.RegisterPlugin;
+
+@RegisterPlugin
+@Requires({
+        @Facet(value = SurfaceHeightFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2)),
+        @Facet(value = BiomeFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2)),
+        @Facet(value = SeaLevelFacet.class, border = @FacetBorder(sides = Igloo.SIZE * 2))
+})
+@Produces(IglooFacet.class)
+public class IglooProvider implements FacetProviderPlugin {
+    private Noise noise;
+
+    @Override
+    public void setSeed(long seed) {
+        noise = new WhiteNoise(seed + 3);
+    }
+
+    /**
+     * Places the Igloo Structure in the Snow Biome with a very low probability of spawn.
+     */
+    @Override
+    public void process(GeneratingRegion region) {
+
+        Border3D border = region.getBorderForFacet(IglooFacet.class).extendBy(0, Igloo.SIZE * 2,
+                Igloo.SIZE * 2);
+
+        IglooFacet facet = new IglooFacet(region.getRegion(), border);
+        SurfaceHeightFacet surfaceHeightFacet = region.getRegionFacet(SurfaceHeightFacet.class);
+        SeaLevelFacet seaLevelFacet = region.getRegionFacet(SeaLevelFacet.class);
+        BiomeFacet biomeFacet = region.getRegionFacet(BiomeFacet.class);
+
+        Rect2i worldRegion = surfaceHeightFacet.getWorldRegion();
+
+        for (int wz = worldRegion.minY(); wz <= worldRegion.maxY(); wz++) {
+            for (int wx = worldRegion.minX(); wx <= worldRegion.maxX(); wx++) {
+                int surfaceHeight = TeraMath.floorToInt(surfaceHeightFacet.getWorld(wx, wz));
+                int seaLevel = seaLevelFacet.getSeaLevel();
+                // check if height is within this region
+                if (surfaceHeight >= facet.getWorldRegion().minY() &&
+                        surfaceHeight <= facet.getWorldRegion().maxY()) {
+                    // Sea Level + 96 is the height at which snow blocks are placed in all biomes.
+                    if (noise.noise(wx, wz) > 0.9999f && (surfaceHeight >= seaLevel + 96f ||
+                            biomeFacet.getWorld(wx, wz).getId().equals(CoreBiome.SNOW))) {
```

Comment:
So the `SNOW` biome always qualifies for placement, that sounds good 👍

Reply:
Yeah, after some testing I felt it was difficult to find an igloo because of the probability anyway, and snow blocks themselves are really high up on the mountains, so the player won't end up stumbling into one very easily unless extremely lucky.
And after seeing the igloos placed while flying around in the biome, the placement felt just what I wanted it to be, sometimes on plain areas and sometimes hidden in the mountains.. giving a proper abandoned feel to them..
Ones in with mountain blocks above the igloo(mini cave like thingies) could've been a result of landslides, etc over a whole lot of years.. story wise...